stdlib/public/core/UTF16.swift

```diff
@@ -434,3 +434,253 @@ extension Unicode.UTF16.ForwardParser: Unicode.Parser, _UTFParser {
     return r
   }
 }
+
+private enum ScalarFallbackResult: UInt8 {
+  case invalid
+  case singleByte
+  case multiByte
+}
+
+#if arch(arm64_32)
+typealias Word = UInt64
+#else
+typealias Word = UInt
+#endif
+let mask = Word(truncatingIfNeeded: 0x80808080_80808080 as UInt64)
+
+#if (arch(arm64) || arch(arm64_32))// && SWIFT_STDLIB_ENABLE_VECTOR_TYPES
```

Comment:
We should do an x86 version of this, but it was giving me weird errors and I wanted to get the core thing working before dealing with them. TBH it's probably still faster on x86 than it was even this way though.